File: charactercounter.js
function countAllCharacters(stringOfWords) {
  // your code here
  var stringcount = {}

  if (stringOfWords == "") {
    return {};
  }

  var words = stringOfWords;
  for (var x = 0; x < words.length; x++) {
    if (stringcount[words[x]] == undefined) {
      stringcount[words[x]] = 1

    }
    else {
      stringcount[words[x]]++;
    }
  }
  return stringcount;
}

var result1 = countAllCharacters('banana');
console.log('should log "{b: 1, a: 3, n: 2}":', "my result:", result1);

var result2 = countAllCharacters('');
console.log('should log "{}":', "my result:", result2);

File: charactercounter.py
def countAllCharacters(stringOfWords):
    stringcount = {}
    if (stringOfWords == ""):
       return {}
    i = 0
    for x in stringOfWords:
        if stringcount[stringOfWords[i]] == KeyError:
            stringcount[f"{stringOfWords[i]}"] = 1
            i = i +1
        else:
            stringcount[f"{stringOfWords[i]}"] = 1
            i = i +1
    return stringcount

result1 = countAllCharacters('banana')
print('should log "{b: 1, a: 3, n: 2}":', "my result:", result1)

result2 = countAllCharacters('')
print('should log "{}":', "my result:", result2);
